hw8/todolist/src/components/Tasks/Tasks.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect } from "react";
+import TaskComponent from "../Task/TaskComponent";
+import { Task } from "../../interfaces/Task";
+
+import { useSelector, useDispatch } from "react-redux";
+import { fetchTasks, deleteTask, completeTask } from "../../taskSlice";
+import { RootState } from "../../store";
+
+interface Props {
+  isCompleted: boolean;
+}
+
+const Tasks = ({ isCompleted }: Props) => {
+  const tasks = useSelector((state: RootState) => state.tasks.tasks);
+  const dispatch = useDispatch();
+
+  const handleComplete = (task: Task) => {
+    dispatch(completeTask(task));
+  };
+
+  const handleDeleteTask = async (deletedTask: Task) => {
+    await fetch(`http://localhost:3004/tasks/${deletedTask.id}`, {
+      method: "DELETE",
+    });
+    dispatch(deleteTask(deletedTask));
+  };
+
+  useEffect(() => {
+    dispatch(fetchTasks());
```

Comment:
You are making the request to server every time you render this component (and you have 2 components that calls the same requests). But you are using redux, so you can call it one time in the parent component and than via the redux just get the data from it. But now from the each component you get the data from the store -> send request to server and change the data -> get this data again